app/controllers/users_controller.rb

```diff
@@ -1362,7 +1362,10 @@ def feature_topic
     user = fetch_user_from_params
     topic = Topic.find(params[:topic_id].to_i)
 
-    raise Discourse::InvalidAccess.new unless topic && guardian.can_feature_topic?(user, topic)
+    if !topic || !guardian.can_feature_topic?(user, topic)
```

Comment:
Very minor suggestion - the guardian could check if `topic.nil?` so you are only checking one condition here, and if in the future someone else uses the guardian method it'll return false for this case too.